kroxylicious-test-tools/src/test/java/io/kroxylicious/test/tester/KroxyliciousClientsTest.java

```diff
@@ -0,0 +1,186 @@
+    void shouldCreateConsumer(Supplier<Consumer<String, String>> consumerSupplier) {
+        // Given
+
+        // When
+        final Consumer<String, String> consumer = consumerSupplier.get();
+
+        // Then
+        assertThat(consumer).isInstanceOf(Consumer.class).isNotNull();
+    }
+
+    @ParameterizedTest
+    @MethodSource("producers")
+    void shouldCreateProducer(Supplier<Producer<String, String>> producerSupplier) {
+        // Given
+
+        // When
+        final Producer<String, String> producer = producerSupplier.get();
+
+        // Then
+        assertThat(producer).isInstanceOf(Producer.class).isNotNull();
+    }
+
+    @Test
+    void shouldCreateAdmin() {
+        // Given
+
+        // When
+        final Admin admin = kroxyliciousClients.admin();
+
+        // Then
+        assertThat(admin).isInstanceOf(Admin.class).isNotNull();
+    }
+
+    @Test
+    void shouldCloseAdminClient() {
+        // Given
+        final Admin admin = kroxyliciousClients.admin();
+
+        // When
+        kroxyliciousClients.close();
+
+        // Then
+        verify(admin).close();
+    }
+
+    @ParameterizedTest
+    @MethodSource("producers")
+    void shouldCloseProducer(Supplier<Producer<String, String>> producerSupplier) {
+        // Given
+        final ProducerRecord<String, String> record = new ProducerRecord<>("topic", "Value");
+        final Producer<String, String> producer = producerSupplier.get();
+
+        // When
+        kroxyliciousClients.close();
+
+        // Then
+        assertThatThrownBy(() -> producer.send(record))
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessage("MockProducer is already closed.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("consumers")
+    void shouldCloseConsumer(Supplier<Consumer<String, String>> consumerSupplier) {
+        // Given
+        final Consumer<String, String> consumer = consumerSupplier.get();
+
+        // When
+        kroxyliciousClients.close();
+
+        // Then
+        assertThatThrownBy(() -> consumer.poll(Duration.ZERO))
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessage("This consumer has already been closed.");
+    }
+
+    @Test
+    void shouldCloseMultipleProducers() {
+        // Given
+        final ProducerRecord<String, String> record = new ProducerRecord<>("topic", "Value");
+        final Set<Producer<String, String>> producers = Set.of(kroxyliciousClients.producer(), kroxyliciousClients.producer(), kroxyliciousClients.producer());
+
+        // When
+        kroxyliciousClients.close();
+
+        // Then
+        for (Producer<String, String> producer : producers) {
+            assertThatThrownBy(() -> producer.send(record))
+                    .isInstanceOf(IllegalStateException.class)
+                    .hasMessage("MockProducer is already closed.");
+        }
+    }
+
+    @Test
+    void shouldCloseMultipleConsumers() {
```

Comment:
I think we are missing a shouldCloseMultipleAdmins test

Reply:
resolved with 16a9f953107d232e33a7a7c814c897956289cdec